what-is-chalk.mdx

```diff
@@ -18,215 +18,467 @@ import { PyDiffEditor } from '@/components/Editor'
-    max_staleness={User.credit_report: "1m"}
-)
 ```
+Using '--out=user'
+Results
+https://chalk.ai/environments/{environment}/query-runs/{query_id}
+Branch: elvis
+Environment: {environment}
+
+ Name                                             Hit?  Value
+─────────────────────────────────────────────────────────────────────────────────────────────────────
+ user.average_rating_given                  4.3125
+ user.birthday                              "1987-12-28"
+ user.created_at                            "2024-08-07T20:54:42.942294+00:00"
+ user.email                                 "98178ad58fe8481db74996996d6e8de7@google.ru"
+ user.first_name                            "Marseda"
+ user.id                                    47
+ user.last_name                             "Karkaletsis"
+ user.review_count                          32
+ user.total_orders_placed                   48
+ user.unique_products_inquired_about        203
+```
+
+Once validated, promote the deployment to production, which is served at a unique preview URL.
 
-### Batch ETL ingestion
-
-Chalk also makes it simple to generate training sets from data warehouse
-sources -- join data from services like S3, Redshift, BQ, Snowflake
-or other custom sources with historical features computed online.
-Specify a cron schedule on an `@offline` resolver, and Chalk will automatically ingest
-data with support for incremental reads:
-
-```sql
--- resolves: User
--- source: snowflake
--- cron: 1h
--- incremental:
---   lookback_period: 10m
---   incremental_column: ts
-select
-    id,
-    credit_score as start_of_day_balance,
-    ts
-from credit_scores
+```shell
+chalk apply
 ```
 
-Chalk makes this data available for point-in-time-correct dataset
-generation for data science use cases. Every pipeline has built-in
-monitoring and alerting to ensure data quality and timeliness.
+These features are now available for low-latency [online inference](/docs/query-basics) and [offline training](/docs/training-client).
+The ability to re-use the same features in both contexts dramatically shortens development time and ensures that feature values from online and offline contexts match.
+
+---
+
+## Why Chalk?
+
+Chalk is the first platform to unify feature and prompt engineering, LLM evals, and real-time inference into a unified platform.
+Bridge the gap between experimentation and production with a single system that unifies offline and online inference.
+Chalk retrieves and computes dynamically at inference time with an execution engine called [Velox](https://github.com/facebookincubator/velox)--we maintain a fork that’s been heavily optimized for low-latency (< 3 ms).
+
+Unifying online and offline execution enables teams to
+
+- consolidate training and serving into a single version controlled environment
+- reduce time-to-production for new models from months to days
+- build advanced AI systems (RAG, RecSys, Fraud detection, etc.) without building custom infrastructure
```

Comment:
capitalization of the starts of the bullets. I don't think Fraud should get capitalization here, b/c it's not an abbreviation? i also have mixed feelings about RAG with these other two -- RAG is a product or strategy, where the other two are industries

Reply:
fixed!